### PR TITLE
v6: Move <Audio*> components to extras

### PR DESCRIPTION
**Pull request by jpaquim**

### CODE REVIEW

File: packages/extras/src/lib/audio/Audio/Audio.svelte [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  if (!listener) {
+    throw new Error(`No Audiolistener with id ${id} found.`)
+  }
+
+  export const audio = new ThreeAudio(listener)
+</script>
+
+<Three
+  type={audio}
+  {position}
```
Comment (grischaerbe) [inline packages/extras/src/lib/audio/Audio/Audio.svelte:49]: use `{...$$restProps} instead of {position}, etc.`

Comment (grischaerbe) [reply packages/extras/src/lib/audio/Audio/Audio.svelte:49]: This includes removing these props from the component script block

File: packages/extras/src/lib/audio/AudioListener/AudioListener.svelte [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  addAudioListener(listener, id)
+
+  onDestroy(() => {
+    removeAudioListener(id)
+  })
+</script>
+
+<Three
+  type={listener}
+  {position}
```
Comment (grischaerbe) [inline packages/extras/src/lib/audio/AudioListener/AudioListener.svelte:44]: Same here, please remove `{position}` et al and insert these via `{...$$restProps}` and remove these props from the script block

File: packages/extras/src/lib/audio/AudioListener/AudioListener.svelte.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import type { Props } from '@threlte/core'
+import type { SvelteComponentTyped } from 'svelte'
+import type { AudioListener as ThreeAudioListener } from 'three'
+
+export type AudioListenerProps = Props<ThreeAudioListener> & {
+  inViewport?: boolean
```
Comment (grischaerbe) [inline packages/extras/src/lib/audio/AudioListener/AudioListener.svelte.d.ts:6]: viewport awareness will be removed in v6 and inserted back via a plugin.

File: packages/extras/src/lib/audio/PositionalAudio/PositionalAudio.svelte [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        directionalCone.coneOuterAngle,
+        directionalCone.coneOuterGain
+      )
+    }
+  }
+</script>
+
+<Three
+  type={audio}
+  {position}
```
Comment (grischaerbe) [inline packages/extras/src/lib/audio/PositionalAudio/PositionalAudio.svelte:67]: Same here, please remove `{position}` et al and insert these via `{...$$restProps}` and remove these props from the script block

File: packages/extras/src/lib/hooks/useAudioListener.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import type { AudioListener } from 'three'
-import { useThrelteAudio } from './useThrelteAudio'
+import { useThrelteAudio } from '@threlte/core'
```
Comment (grischaerbe) [inline packages/extras/src/lib/hooks/useAudioListener.ts:2]: The audio context and its hook should be moved to extras as well.

Comment (grischaerbe) [reply packages/extras/src/lib/hooks/useAudioListener.ts:2]: The problem then is: what component is providing that context? Will there be a component `<Extras>` that you wrap your entire scene in that provides all necessary contexts used in `@threlte/extras`? Is this a use case for a root plugin?

**Review by grischaerbe — CHANGES_REQUESTED** [review]
Thanks @jpaquim for your PR!
I added some comments, for some I'm not sure how to handle it concerning the contexts, others are clear.

File: packages/extras/src/lib/audio/AudioListener/AudioListener.svelte.d.ts (hunk 3 above)
Comment (jpaquim) [reply packages/extras/src/lib/audio/AudioListener/AudioListener.svelte.d.ts]: Removed viewport-related props and events

**Comment by grischaerbe** [comment]
The deployment fails because of this error: 

```
@threlte/docs-next:build: "useAudioListener" is not exported by "../../packages/core/src/lib/index.ts", imported by "src/examples/core/positional-audio/Turntable.svelte".
```

Other than that, is this ready to be merged? 🚀

**Review by grischaerbe — APPROVED** [review]
LGTM 🚀 
Thanks so much!

**Pull request merged**